modules/swagger-codegen/src/main/resources/typescript-angular/apiInterface.mustache

```diff
@@ -14,9 +23,9 @@ import { Configuration }                                     from '../configurat
     */
 {{/description}}
 export interface {{classname}}Interface {
-    defaultHeaders: Headers;
+    defaultHeaders: {{#useHttpClient}}Http{{/useHttpClient}}Headers;
     configuration: Configuration;
-    [others: string]: any;
+    {{^useHttpClient}}[others: string]: any;{{/useHttpClient}}
```

Comment:
can anyone explain why this is necessary / not necessary anymore?

Reply:
@macjohnny I'm not sure either. The `useHttpClient` tag is introduced for Typescript version 4.3 or later: https://github.com/swagger-api/swagger-codegen/blob/157e6b7fab4c4b3cddee88fb1100271c2365a6da/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/TypeScriptAngularClientCodegen.java#L127

Looks like that line is needed for Angular 2.x but not Angular 4.3 or later?